cmd/addblock/import.go

```diff
@@ -295,9 +296,40 @@ func (bi *blockImporter) Import() chan *importResults {
 // newBlockImporter returns a new importer for the provided file reader seeker
 // and database.
 func newBlockImporter(db database.DB, r io.ReadSeeker) (*blockImporter, error) {
+	// Create the transaction and address indexes if needed.
+	//
+	// CAUTION: the txindex needs to be first in the indexes array because
+	// the addrindex uses data from the txindex during catchup.  If the
+	// addrindex is run first, it may not have the transactions from the
+	// current block indexed.
+	var indexes []indexers.Indexer
+	if cfg.TxIndex || cfg.AddrIndex {
+		// Enable transaction index if address index is enabled since it
+		// requires it.
+		if !cfg.TxIndex {
+			log.Infof("Transaction index enabled because it is " +
+				"required by the address index")
+			cfg.TxIndex = true
+		} else {
+			log.Info("Transaction index is enabled")
+		}
+		indexes = append(indexes, indexers.NewTxIndex(db))
+	}
+	if cfg.AddrIndex {
+		log.Info("Address index is enabled")
```

Comment:
That's doable, although I think you would need the extra check one way or the other in order for the logging message to say when the `txindex` has been enabled due to the `address` index versus due to the config setting.  The way it is now, the transaction index state is always listed before the address index state. I sort of like showing the index that is a dependency first, but I don't guess it would matter much to reverse them.

For example, the reversed approach might look like:

```Go
var forcedTxIndex bool
if cfg.AddrIndex {
	forcedTxIndex = true
	cfg.TxIndex = true
	log.Info("Address index is enabled")
	indexes = append(indexes, indexers.NewAddrIndex(db, activeNetParams))
}
if cfg.TxIndex {
	if forcedTxIndex {
		log.Infof("Transaction index enabled because ...")
	} else {
		log.Info("Transaction index is enabled")
	}
	indexes = append(indexes, indexers.NewTxIndex(db))
}
```

There isn't much difference to my eyes.